load-tsfile-tool/src/main/java/org/apache/iotdb/TsFileLoaderTool.java

```diff
@@ -0,0 +1,149 @@
+    Options options = createOptions();
+    try {
+      CommandLine commandLine = new DefaultParser().parse(options, args);
+      host = getArgOrDefault(commandLine, "h", host);
+      port = getArgOrDefault(commandLine, "p", port);
+      user = getArgOrDefault(commandLine, "u", user);
+      password = getArgOrDefault(commandLine, "pw", password);
+      filePath = getArgOrDefault(commandLine, "f", filePath);
+    } catch (ParseException e) {
+      System.out.println(String.format("Parse Args Error. %s", e.getMessage()));
+      priHelp(options);
+    }
+  }
+
+  private static void priHelp(Options options) {
+    new HelpFormatter().printHelp("./load-tsfile.sh(load-tsfile.bat if Windows)", options, true);
+  }
+
+  private static String getArgOrDefault(
+      CommandLine commandLine, String argName, String defaultValue) {
+    String value = commandLine.getOptionValue(argName);
+    return value == null ? defaultValue : value;
+  }
+
+  public static Options createOptions() {
+    Options options = new Options();
+    Option help = new Option("help", false, "Display help information(optional)");
+    help.setRequired(false);
+    options.addOption(help);
+
+    Option host =
+        Option.builder("h")
+            .argName("host")
+            .hasArg()
+            .desc("Host Name (optional, default 127.0.0.1)")
+            .build();
+    options.addOption(host);
+
+    Option port =
+        Option.builder("p").argName("port").hasArg().desc("Port (optional, default 6667)").build();
+    options.addOption(port);
+
+    Option username =
+        Option.builder("u")
+            .argName("username")
+            .hasArg()
+            .desc("User name (required)")
+            .required()
+            .build();
+    options.addOption(username);
+
+    Option password =
+        Option.builder("pw").argName("password").hasArg().desc("password (optional)").build();
+    options.addOption(password);
+
+    Option filePathOpt =
+        Option.builder("f")
+            .argName("file")
+            .hasArg()
+            .desc("File or Dictionary to be loaded.")
+            .required()
+            .build();
+    options.addOption(filePathOpt);
+    return options;
+  }
+
+  public static List<File> collectTsFiles(File file) {
+    if (file.isFile()) {
+      return file.getName().endsWith(TsFileConstant.TSFILE_SUFFIX)
+          ? Collections.singletonList(file)
+          : Collections.emptyList();
+    }
+    List<File> list = new ArrayList<>();
+    for (File listFile : file.listFiles()) {
+      if (listFile.isDirectory()) {
+        list.addAll(collectTsFiles(listFile));
+      } else if (listFile.getName().endsWith(TsFileConstant.TSFILE_SUFFIX)) {
+        list.add(listFile);
+      }
+    }
+    return list;
+  }
+
+  public static void writeToIoTDB(List<File> files, Session session) {
+    for (File file : files) {
+      RewriteFileTool.rewriteWrongTsFile(file.getPath(), session);
```

Comment:
if some file is corrupt, we need to rewrite all tsfiles again? Could we log it and support restart from the failed file?

Reply:
you can find one of solution in [user guide](https://apache-iotdb.feishu.cn/docx/doxcntEPZnXTGtQcdTk83yuOSqh), i give the list of tsfile which are already loaded, so people can find the tsfiles which do not need to load again.